src/libluau.zig

```diff
@@ -961,6 +972,24 @@ pub const Lua = struct {
         return error.Fail;
     }
 
+    /// Converts the Lua string at the given `index` to a string atom.
+    /// The Lua value must be a string.
+    pub fn toStringAtom(lua: *Lua, index: i32, atom: *c_int) ![*:0]const u8 {
+        if (c.lua_tostringatom(lua.state, index, atom)) |ptr| {
+            return ptr;
+        }
+        return error.Fail;
+    }
+
+    /// Retrieve the user atom index and name for the method being
+    /// invoked in a namecall.
+    pub fn namecallAtom(lua: *Lua, atom: *c_int) ![*:0]const u8 {
+        if (c.lua_namecallatom(lua.state, atom)) |ptr| {
+            return ptr;
+        }
+        return error.Fail;
+    }
```

Comment:
All good ideas! I haven't used the destructuring syntax yet but this is a good candidate for that and also hides the use of c_int (if that's deemed safer than i32, although latter most likely is fine too). 

Agreed on std.mem.span although it may make sense to retain [:0] in the type since it's a pain to get it back if needed for some reason. 🙂

I'll update the PR when I'm back at the keyboard.

Reply:
Great review!  The destructuring syntax is great.  Using the `namecallAtom` looks pretty clean now:

```
        pub fn vectorNamecall(l: *Lua) i32 {
            const atom_idx, _ = l.namecallAtom() catch {
                l.raiseErrorStr("%s is not a valid vector method", .{l.checkString(1)});
            };
            switch (atom_idx) {
                dot_idx => {
                    const a = l.checkVector(1);
                    const b = l.checkVector(2);
                    l.pushNumber(a[0] * b[0] + a[1] * b[1] + a[2] * b[2]); // vec3 dot
                    return 1;
                },
                sum_idx => {
                    const a = l.checkVector(1);
                    l.pushNumber(a[0] + a[1] + a[2]);
                    return 1;
                },
                else => unreachable,
            }
        }
```

Updated the PR with a new commit.  I'm not sure which way you prefer, new commits or squash+force push.  I generally prefer to merge squashed PRs but reviewing is nicer this way.

Reply:
Forgot to mention that I think it's better to use i32 than i16 for atom indices.  That's what the Luau API uses too.

Reply:
Squashed and updated.  Good to go again, thanks for the review!

i16 is involved (the user atom callback returns i16) but I chose to keep the word lengths the same as in the Luau API.